src/CSharp/CodeCracker/Style/RemoveAtFromVariablesAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace CodeCracker.CSharp.Style
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class RemoveAtFromVariablesAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Remove @ from variables that are not keywords.";
+        internal const string MessageFormat = "Remove @ from variables that are not keywords.";
+        internal const string Category = SupportedCategories.Style;
+        const string Description = "Usage of @ on variable names only when it is a CSharp keyword";
+
+        internal static readonly DiagnosticDescriptor RuleNonPrimitives = new DiagnosticDescriptor(
+            DiagnosticId.RemoveAtFromVariablesThatAreNotKeywords.ToDiagnosticId(),
+            Title,
+            MessageFormat,
+            Category,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true,
+            description: Description,
+            helpLinkUri: HelpLink.ForDiagnostic(DiagnosticId.RemoveAtFromVariablesThatAreNotKeywords));
+
+        internal static readonly DiagnosticDescriptor RulePrimitives = new DiagnosticDescriptor(
```

Comment:
I was learning using another Analyzer as an example, probably I forgot to remove. I'll fix that at night.